packages/opentelemetry-types/src/trace/trace_state.ts

```diff
@@ -14,6 +14,40 @@
  * limitations under the License.
  */
 
+/**
+ * Tracestate carries system-specific configuration data, represented as a list
+ * of key-value pairs. TraceState allows multiple tracing systems to
+ * participate in the same trace.
+ */
 export interface TraceState {
-  // TODO
+  /**
+   * Adds or updates the TraceState that has the given `key` if it is
+   * present. The new State will always be added in the front of the
+   * list of states.
+   *
+   * @param name key of the TraceState entry.
+   * @param value value of the TraceState entry.
+   */
+  set(name: string, value: string): void;
+
+  // TODO: Consider to add support for merging an object as well by also
+  // accepting a single internalTraceState argument similar to the constructor.
+
+  /**
+   * Serializes the TraceState to a `list` as defined below. The `list` is a
+   * series of `list-members` separated by commas `,`, and a list-member is a
+   * key/value pair separated by an equals sign `=`. Spaces and horizontal tabs
+   * surrounding `list-members` are ignored. There can be a maximum of 32
+   * `list-members` in a `list`.
+   *
+   * @returns the serialized string.
+   */
+  serialize(): string;
+
+  /**
+   * Parses the input string and generates a TraceState.
```

Comment:
The documentation isn't entirely correct as it returns void, should change to `Parses the input string and adds it to the TraceState`, although that also isn't correct since it actually overrides the internal trace state in implementation. I feel like `parse` should probably go back to being a standalone or static function that returns a new `TraceState`, the constructor for `TraceState` can accept either a string or InternalTraceState.

From a code point of view, either of these seems ideal
```javascript
rawTraceState = req.headers.get("tracestate"); 
traceState = new TraceState(rawTraceState); 
// or something like this (parse returns a trace state)
traceState = tracestate.parse(rawTraceState);
// but allowing for this to happen looks smelly
traceState = new TraceState();
traceState.parse(rawTraceState); // actually sets the internal state
```
thoughts?

Reply:
If `TraceState.parse` would return a new `TraceState`, I think this means it is simply duplicating the constructor, in which case I would be in favor of removing it completely.

Reply:
Done in https://github.com/open-telemetry/opentelemetry-js/pull/57/commits/c86a044efb589975b85e30f973f36b28ecb5090a